packages/twenty-front/src/modules/object-record/utils/computeOptimisticRecordFromInput.ts

```diff
@@ -0,0 +1,144 @@
+import { isNull, isString, isUndefined } from '@sniptt/guards';
+
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import {
+  getRecordFromCache,
+  GetRecordFromCacheArgs,
+} from '@/object-record/cache/utils/getRecordFromCache';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { RelationDefinitionType } from '~/generated-metadata/graphql';
+import { FieldMetadataType } from '~/generated/graphql';
+import { isDefined } from '~/utils/isDefined';
+import { getUrlHostName } from '~/utils/url/getUrlHostName';
+
+// We're not strict checking that any key within recordInput exists in the objectMetadaItem
+type ComputeOptimisticCacheRecordInputArgs = {
+  objectMetadataItem: ObjectMetadataItem;
+  recordInput: Partial<ObjectRecord>;
+} & Pick<GetRecordFromCacheArgs, 'cache' | 'objectMetadataItems'>;
+export const computeOptimisticRecordFromInput = ({
+  objectMetadataItem,
+  recordInput,
+  cache,
+  objectMetadataItems,
+}: ComputeOptimisticCacheRecordInputArgs) => {
+  const accumulator: Partial<ObjectRecord> = {};
+  for (const fieldMetadataItem of objectMetadataItem.fields) {
+    const isUuidField = fieldMetadataItem.type === FieldMetadataType.UUID;
+    if (isUuidField) {
+      // Could be a util ?
+      const isRelationFieldId = objectMetadataItem.fields.some(
+        ({ type, relationDefinition }) => {
+          if (type !== FieldMetadataType.RELATION) {
+            return;
+          }
+
+          if (!isDefined(relationDefinition)) {
+            return;
+          }
+
+          const sourceFieldName = relationDefinition.sourceFieldMetadata.name;
+          // TODO we should create utils to centralized this logic
+          return `${sourceFieldName}Id` === fieldMetadataItem.name;
+        },
+      );
+
+      if (isRelationFieldId) {
+        continue;
+      }
+    }
+
+    const isRelationField =
+      fieldMetadataItem.type === FieldMetadataType.RELATION;
+
+    if (
+      isRelationField &&
+      fieldMetadataItem.relationDefinition?.direction ===
+        RelationDefinitionType.ONE_TO_MANY
+    ) {
+      continue;
+    }
+
+    const isManyToOneRelation =
+      fieldMetadataItem.relationDefinition?.direction ===
+      RelationDefinitionType.MANY_TO_ONE;
+    const recordInputFieldValue: unknown = recordInput[fieldMetadataItem.name];
```

Comment:
unknown should be string | null

Reply:
check isFieldRelationToOneValue if the typing is an issue

Reply:
Unless I'm mistaken it should not, because it could be any other field mutation from any ObjectRecord fields ?